speech/src/test/java/com/example/speech/InfiniteStreamRecognizeTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.speech;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.cloud.speech.v1p1beta1.RecognitionConfig;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import javax.sound.sampled.AudioFormat;
+import javax.sound.sampled.AudioInputStream;
+import javax.sound.sampled.AudioSystem;
+import javax.sound.sampled.UnsupportedAudioFileException;
+import org.junit.Before;
+import org.junit.Test;
+
+public class InfiniteStreamRecognizeTest {
+  private static final String COMMERCIAL_MONO_FILE = "resources/commercial_mono.wav";
+  private ByteArrayOutputStream bout;
+
+  @Before
+  public void setUp() throws IOException, ExecutionException, InterruptedException,
+          TimeoutException {
+    bout = new ByteArrayOutputStream();
+    PrintStream out = new PrintStream(bout);
+    System.setOut(out);
+  }
+
+  @Test
+  public void infiniteStreamingRecognizeTest() throws Exception {
+    InfiniteStreamRecognize.putDataToSharedQueue(readAudioFile());
+    InfiniteStreamRecognize.putDataToSharedQueue("exit".getBytes(StandardCharsets.UTF_8));
+    InfiniteStreamRecognize.infiniteStreamingRecognize("en-US",
+            () -> {}, 16000, RecognitionConfig.AudioEncoding.LINEAR16);
+    // wait responses from server
+    Thread.sleep(10000);
+    assertThat(bout.toString().toLowerCase()).contains("hi i want to");
+    assertThat(bout.toString().toLowerCase()).contains("recognition was stopped");
```

Comment:
I listen to the commercial_mono.wav file and it does not contain a word "exit". Does it mean that `onComplete()` is called when the stream ends _OR_ stopped?
I recommend to consider distinguishing these two events. Currently the impression from the code sample is that onComplete() is called when the recognition is stopped.

Reply:
As we have InfiniteStreamRecognize, I suppose we don't have some status-completed, so this code can call `onComplete()` only in one way when we stop the recognition

Reply:
Integration tests are supposed to be deterministic, terminated, run minimal time required to execute the test and do a proper setup and clean up action so they aren't dependent on the state of the testing environment and do not consume resources beyond required.
From your comment it sounds like the test might not terminate, making it a subject for flakiness.

Please, reconsider the testing strategy because we prefer not to have a test at all to having a flaky test.